### ISSUE TITLE
How to make it responsible

### ISSUE THREAD

**Issue by zefexdeveloper** [Norserium/vue-advanced-cropper#28]
I'm planning in using the cropper on a modal, the problem I'm having is that even tho the width is being respected of the modal, the height isn't. If the user selects an image bigger than the height of the browser it will grow vertically and eventually disappears.

How can I make so the cropper respects the width and height like keeping its aspect ratio based on the width of the modal?

![image](https://user-images.githubusercontent.com/30276092/70849602-8a587200-1e5f-11ea-8f61-f80b411c9785.png)

It's more or less like this example: https://norserium.github.io/vue-advanced-cropper/tutorials/recipes.html#upload-image

If you select a REALLY big height image it won't get bigger, it will scale to fit the area.

An image for example: https://picsum.photos/id/96/800/2000

**Comment by Norserium** [comment]
Hello, @zefexdeveloper! Did you try to set max height for the cropper or its wrapper?

**Comment by zefexdeveloper** [comment]
@Norserium Yes, after inspecting your example I saw that you set the `max-height` as `500`. The thing is that I want to put that in a modal and I don't know if setting a fixed `max-height` would be essencial for responsiveness, like with smaller devices.
I'm a backend developer so I find it really complicated when it comes to front-end and make things responsive haha.

**Comment by Norserium** [comment]
What's the library do you use for modal window? I suppose you can set for example `max-height: calc(100vh - 40px)`.

**Comment by zefexdeveloper** [comment]
I'm using tailwindcss and yes, for things like this I can always use inline `style`, I won't be using this in another place, otherwise I would create a reusable class.
I will be trying that out.

**Comment by Norserium** [comment]
Hello, @zefexdeveloper! Did you solve your problem?

**Comment by zefexdeveloper** [comment]
Yes, kinda, I dropped the modal, it was giving me headaches, I went by switching components and showing the cropping when the user selects the image instead.

**Comment by Norserium** [comment]
Was my advice not helpful?

**Comment by zefexdeveloper** [comment]
I'm sorry the delay answering. Yes it did help, thank you very much. ♥

**Comment by zefexdeveloper** [comment]
@Norserium 

I went back to this issue cause I'm trying to set max height as you did you on your upload example:

![image](https://user-images.githubusercontent.com/30276092/71530517-dcba7980-28c8-11ea-8897-f69a5e99689c.png)

I tried to do the same thing but it doesn't work, it's not respecting the max height:

![image](https://user-images.githubusercontent.com/30276092/71530577-1ee3bb00-28c9-11ea-85b9-a4ef14c5ec28.png)

```
    <div class="relative">
        <div class="relative">
            <div class="border border-gray-200 min-h-300px max-h-500px">
                <cropper ref="cropper" :src="image" @ready="loading = false"></cropper>
            </div>
        </div>

        <div class="flex justify-center mt-10">
            <button class="ml-2 btn btn-gold" @click.prevent="crop">Crop</button>
        </div>
    </div>
```
The classes are from `Tailwindcss` utility classes. (it doesn't work even if I set using `style` - meaning it's not a `Tailwind` problem).

**Comment by Norserium** [comment]
It's strange. Could you try to set `max-height` for the cropper itself, not its wrapper?

**Comment by zefexdeveloper** [comment]
![image](https://user-images.githubusercontent.com/30276092/71532069-70437880-28d0-11ea-8612-c41cd40d87d5.png)

You are right, it's working now. Thank you.

One thing left tho. I have a spinner that I set using the `ready` event that the cropper returns when it's done loading, the problem is that the controls are there anyway (2nd image)

![image](https://user-images.githubusercontent.com/30276092/71532101-9406be80-28d0-11ea-8696-6f6618c7ba2a.png)

This cursor shows, meaning that the controls appears even when loading.
![image](https://user-images.githubusercontent.com/30276092/71532192-05467180-28d1-11ea-9409-7339c2b64925.png)

How can I only show the cropper itself or the controls when it's literally ready?

I tried adding `v-show="! loading"` (`loading` is set as `false` when the cropper fires `ready`) but it doesn't work.

**Comment by Norserium** [comment]
I see. It's the bug obliviously, I will fix later.

**Comment by zefexdeveloper** [comment]
@Norserium Yes, the controls like for resizing is appearing even when the cropper is loading (the `ready` is not fired) so when I show like a spinner if the user mouses over the cropper space he can see the cursor changing to the resize one as the controls are there already before the image is loaded.

**Comment by Norserium** [comment]
@zefexdeveloper, this bug is fixed in 0.15.2

**Comment by zefexdeveloper** [comment]
@Norserium You rock, thank you for that.
